node/silkworm/rpc/call.hpp

```diff
@@ -479,12 +489,16 @@ class BidirectionalStreamingRpc : public BaseRpc {
         SILK_TRACE << "BidirectionalStreamingRpc::BidirectionalStreamingRpc END new request issued [" << this << "]";
     }
 
+    virtual void start() = 0;
```

Comment:
Is this a method pertaining only to BidirectionalStreamingRpc and to its descendants or to all Rpc? The same for end() and process().

Reply:
The `start/end` methods are meaningful only for bidirectional-streaming calls (and probably client-streaming ones, but we don't need to support them ATM) because they can have more than one incoming request from the same client, hence it makes sense to have hooks for call start, for each request processing and for call end.

The `process` method is indeed present in all RPC types we support (unary, server-streaming, bidirectional-streaming) but the call sites for such an hook is different in each RPC type, i.e. there is no place in code where you have a pointer or reference to `BaseRpc` and you need to upcall `process`. This is the reason it is not defined at the base class level.